src/utilities/strategy-helpers/blue-green-helper.ts

```diff
@@ -0,0 +1,822 @@
+'use strict';
+
+import { Kubectl } from '../../kubectl-object-model';
+import * as kubectlUtils from '../kubectl-util';
+import * as core from '@actions/core';
+import * as fs from 'fs';
+import * as yaml from 'js-yaml';
+import { checkForErrors } from '../utility';
+import * as util from 'util';
+
+import * as fileHelper from '../files-helper';
+import * as helper from '../resource-object-utility';
+import * as TaskInputParameters from '../../input-parameters';
+
+export const BLUE_GREEN_NEW_LABEL_VALUE = 'green';
+export const NONE_LABEL_VALUE = 'None';
+export const BLUE_GREEN_DEPLOYMENT_STRATEGY = 'BLUE-GREEN';
+export const BLUE_GREEN_VERSION_LABEL = 'k8s.deploy.color';
+export const BLUE_GREEN_SUFFIX = '-green';
+export const INGRESS_ROUTE = 'INGRESS';
+export const SMI_ROUTE = 'SMI';
+export const STABLE_SUFFIX = '-stable'
+export const TRAFFIC_SPLIT_OBJECT_NAME_SUFFIX = '-rollout';
+export const TRAFFIC_SPLIT_OBJECT = 'TrafficSplit';
+export const MIN_VAL = '0';
+export const MAX_VAL = '100';
+
+import { KubernetesWorkload } from '../../constants';
+import { StringComparer, isEqual } from '../string-comparison';
+
+let trafficSplitAPIVersion = "";
+
+// service implementation
+export function isBlueGreenDeploymentStrategy() {
+    const deploymentStrategy = TaskInputParameters.deploymentStrategy;
+    return deploymentStrategy && deploymentStrategy.toUpperCase() === BLUE_GREEN_DEPLOYMENT_STRATEGY;
+}
+
+export function deployBlueGreen(kubectl: Kubectl, filePaths: string[]) {
+
+    // get all kubernetes objects defined in manifest files
+    const manifestObjects = getManifestObjects(filePaths);
+
+    // checking if there are existing green deployments with green label value
+    shouldWeDeploy(kubectl, manifestObjects.depEntList);
+
+    // create deployments with green label value
+    const result = createDeploymentsWithLabel(kubectl, manifestObjects.depEntList, BLUE_GREEN_NEW_LABEL_VALUE);
+
+    // create other non deployment and non service entities
+    const newObjectsList = manifestObjects.otherObjects.concat(manifestObjects.ingEntList);
+    const manifestFiles = fileHelper.writeObjectsToFile(newObjectsList);
+    kubectl.apply(manifestFiles);
+    
+    // returning deployment details to check for rollout stability
+    return result;
+}
+
+export async function blueGreenPromote(kubectl: Kubectl, filePaths: string[]) {
+
+    // get all kubernetes objects defined in manifest files
+    const manifestObjects = getManifestObjects(filePaths);
+
+    // checking if services are in the right state ie. targeting green deployments
+    shouldWePromote(kubectl, manifestObjects.depEntList, manifestObjects.serEntList);
```

Comment:
Change it to `validateServiceState` or you can probably make it a boolean function `isServiceTargettingWorkloads` or something like that. You'll have to add an if-else block here in that case.